pkg/cri/opts/spec_windows.go

```diff
@@ -138,9 +138,18 @@ func WithWindowsMounts(osi osinterface.OS, config *runtime.ContainerConfig, extr
 				// hcsshim requires clean path, especially '/' -> '\'. Additionally,
 				// for the destination, absolute paths should have the C: prefix.
 				src = filepath.Clean(src)
-				dst = filepath.Clean(dst)
-				if dst[0] == '\\' {
-					dst = "C:" + dst
+
+				// filepath.Clean adds a '.' at the end if the path is a
+				// drive (like Z:, E: etc.). Keeping this '.' in the path
+				// causes incorrect parameter error when starting the
+				// container on windows.  Remove it here.
+				if !(len(dst) == 2 && dst[1] == ':') {
+					dst = filepath.Clean(dst)
+					if dst[0] == '\\' {
+						dst = "C:" + dst
+					}
+				} else if dst[0] == 'c' || dst[0] == 'C' {
```

Comment:
- Wouldn't it be applicable for D: and other drive letters as well?
- We probably should have some tests around this function.

Reply:
Agreed lets add some tests. To my knowledge there is no destination on Windows for a container that isn't C:\ rooted right? Should we just fail all mounts that don't have this prefix? Is there some type of mount I am not considering? (I know at the os layer there are but at this layer its just a bind mount right?)

Reply:
Agreed lets add some tests. To my knowledge there is no destination on Windows for a container that isn't C:\ rooted right? Should we just fail all mounts that don't have this prefix? Is there some type of mount I am not considering? (I know at the os layer there are but at this layer its just a bind mount right?)

Reply:
Why is GH double posting...

Reply:
@kzys It shouldn't fail for drive letters other than `C` since you can mount a directory on the host as a drive inside the container. It is just not possible with the `C:` drive because that drive is already used inside the container when container starts. I have added tests. Let me know if it looks good.
@jterry75 As I said above you can mount a directory on the host as a drive inside the container. Docker supports this but containerd doesn't (due to this bug).

Reply:
I missed that thanks for the clarity